### ISSUE TITLE
Sending operation object to stackdriver logging

### ISSUE THREAD

**Issue by kdabir** [GoogleCloudPlatform/fluent-plugin-google-cloud#126]
As per Stackdriver Logging API, we could pass an `operation` object to connect relevant log lines that are part of one operation. (Reference: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#LogEntryOperation)

Here are some example log lines:
```
{"timestamp":"2017-04-21T22:24:33.739+05:30","message":"message 1","logger":"LoggerRequestListener","thread":"http-nio-8080-exec-1","severity":"INFO","level_value":20000,"operation":{"producer":"api","id":"nqTUHtJueuw45jcAVpfU"}}
{"timestamp":"2017-04-21T22:24:33.739+05:30","message":"message 1","logger":"LoggerRequestListener","thread":"http-nio-8080-exec-1","severity":"INFO","level_value":20000,"operation":{"producer":"api","id":"nqTUHtJueuw45jcAVpfU"}}
{"timestamp":"2017-04-21T22:24:33.739+05:30","message":"message 3","logger":"LoggerRequestListener","thread":"http-nio-8080-exec-1","severity":"INFO","level_value":20000,"operation":{"producer":"api","id":"nqTUHtJueuw45jcAVpfU"}}
```

But when viewed in Stackdriver Logging, `operation` is nested inside `jsonPayload`. 

![image](https://cloud.githubusercontent.com/assets/735240/25289650/24c95f06-26e8-11e7-898e-579302b711f5.png)

When used similar feature for `httpRequest`, google fluentd plugin moves the object at top level but `operation` gets no such love from the plugin. 

![image](https://cloud.githubusercontent.com/assets/735240/25289659/340159ec-26e8-11e7-88a0-d88a27fe003b.png)

How do we send `operation` object such that it is received at top level by Stackdriver logging?

**Comment by kdabir** [comment]
any updates on this one? I hope the request makes sense. Let me know if I can help with anything.

**Comment by kdabir** [comment]
I am losing hopes on getting updates on this. If it can not be supported, kindly update the issue, we'll stop watching it.

**Comment by j-walker23** [comment]
It's sad no one has responded to this. This was the number one coolest logging feature anyone i have ever talked to loved. Only thing i truly miss from my old appengine days.

@kdabir I am just starting to play with this stuff now so i have no evidence yet but it seems to me that `operation` isn't what you really want. At least according to this hopefully magical [doc](https://gist.github.com/JustinBeckwith/43dd04cfbc4551ddd8996a5b803da0b0).

[LogEntry](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry) does have a section about `trace` so i am hoping that is the same `trace_id` stuff that you were thinking `operation` was for.

Sadly though, this fluentd plugin will not support any of this according to all of the source code i just read so you'll have to go a different route. I am using python and it looks like `google-cloud-logging` might finally be useable at scale so am going to give all of this a try.

I'll report if i get the magic logging back!

**Comment by igorpeshansky** [comment]
Apologies for the long silence. The request definitely makes sense, and this is on our roadmap, but there is no ETA at the moment. I'll make another update to this issue once we have anything new to report.

**Comment by qingling128** [comment]
Hi @draffensperger - FYI, there is a new pull request (https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/145) that seems to overlap a bit with the `operation` field support you are planning to work on. Mind commenting there on what the current plan/status is?

**Comment by kdabir** [comment]
so I think I need to wait for 0.6.5 release to contain this fix. Is that correct?

**Comment by qingling128** [comment]
@kdabir Yes. I've just released [v0.6.5](https://rubygems.org/gems/fluent-plugin-google-cloud/versions/0.6.5).
